zebrad/src/components/mempool/error.rs

```diff
@@ -0,0 +1,29 @@
+//! Errors that can occur when adding transactions to the mempool or querying it.
+
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq)]
+#[allow(dead_code)]
+pub enum TransactionInsertError {
+    #[error("transaction already exists in mempool")]
+    InMempool,
+
+    #[error("transaction did not pass consensus validation")]
+    TransactionInvalid(#[from] zebra_consensus::error::TransactionError),
```

Comment:
Optional naming style: In Rust, we try to avoid repeating words in the enum type and variant names.

For example: `TransactionInsertError::Invalid`, not `TransactionInsertError::TransactionInvalid`.